src/luascripts/highscore_show.lua

```diff
@@ -1,37 +1,42 @@
-local username = {}
-local skin = {}
-local time = {}
-local full_rows = 0
-local used_row = 0
-local no_time = true
+--Script by Meziu & LeonardoTheMutant
 
-local function receive_data(source, type, target, msg)
-	if msg:sub(0, 5) == "UXDFS"
-		local data_msg = msg:sub(6)
+local username, skin, time, time_str, recieved_row, used_row, got_time
+
+local function SCTCL_ResetBuffer()
+	username = {} --username (string)
+	skin = {} --skin (string)
+	time = {} --time (integer, tics value)
+	time_str = {} --time (string, tics value converted to mm:ss.cc)
+	recieved_row = 0 --the row number being currently procecced by the reciever
+	used_row = -1 --row used by the current player
+end
+
+--
+-- The "SCTCL" reciever
+-- ("Server C To Client Lua")
+--
+addHook("PlayerMsg", function(s, t, r, m) --source, type, reciever (target), message
+	if (m:sub(0, 6) == "#SCTCL") --'#' to make it more rare for players to accidentaly type in a conversation
```

Comment:
an additional check can be added to make sure that the message was sent from the server
```lua
if (m:sub(0, 6) == "#SCTCL") and (s == server)
```